---
title: "cornerstone_its_dada2_github"
author: Benjamin Robinson
output: html_notebook
---

```{r style, echo=FALSE, message=FALSE, warning=FALSE, results="asis"}
library("BiocStyle")
library("knitr")
library("rmarkdown")
options(width = 98) 
opts_chunk$set(message = FALSE, error = FALSE, warning = FALSE,
               cache = TRUE, fig.width = 8, fig.height = 7)
```

-------------------------------------------------
First we load the necessary packages.

```{r, message=FALSE, warning=FALSE}
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
   install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
if(any(!.inst)) {
   source("http://bioconductor.org/biocLite.R")
   biocLite(.bioc_packages[!.inst], ask = F)
}
# Load packages into session, and print package version
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
set.seed(100)
```

```{r path}
miseq_path <- "C:/Users/benja/Desktop/cornerstone_ITS_fastq/try2" # CHANGE to the directory containing the fastq files.
list.files(miseq_path)
```

 Filter and Trim {#trim-and-filter .unnumbered}
-----------------------------------------------
```{r filenames}
# Sort ensures forward/reverse reads are in same order
fnFs <- sort(list.files(miseq_path, pattern="_R1_001.fastq.gz"))
fnRs <- sort(list.files(miseq_path, pattern="_R2_001.fastq.gz"))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)
sampleNames <- sapply(strsplit(sampleNames, "202"), `[`, 1)
# Specify the full path to the fnFs and fnRs
fnFs <- file.path(miseq_path, fnFs)
fnRs <- file.path(miseq_path, fnRs)
fnFs[1:3]
fnRs[1:3]
```

The first two forward reads:
```{r see-quality-F}
plotQualityProfile(fnFs[1:3])
```

The first two reverse reads:
```{r see-quality-R}
plotQualityProfile(fnRs[1:3])
```

We define the filenames for the filtered fastq.gz files:
```{r filt-names}
filt_path <- file.path(miseq_path, "filtered") # Place filtered files in filtered/ subdirectory
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq.gz"))

```

```{r}
sampleNames
```

**Filter the forward and reverse reads**: #these are very standard cutoffs
```{r filter, message=FALSE, warning=FALSE}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, maxN = 0, maxEE = c(2, 2), 
    truncQ = 2, minLen = 50, rm.phix = TRUE, compress = TRUE, multithread = FALSE) # On Windows set multithread=FALSE
head(out)
```

```{r}
plotQualityProfile(filtFs[1:3])
plotQualityProfile(filtRs[1:3])
```


Infer sequence variants {#infer-sequence-variants .unnumbered}
--------------------------------------------------------------
### Dereplication {.unnumbered
```{r dereplicate, message=FALSE}
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames
```

```{r learnErrorRates}
errF <- learnErrors(filtFs)
errR <- learnErrors(filtRs)
```

```{r plotrates, fig.show= "hold", fig.height = 8, fig.cap= "Estimated Error rates (both forward and reverse)"}
plotErrors(errF)
plotErrors(errR)
```

```{r dadaStep}
dadaFs <- dada(derepFs, err=errF, pool="pseudo")
dadaRs <- dada(derepRs, err=errR, pool="pseudo")
```

Inspecting the dada-class object returned by dada:
```{r see-dada}
dadaFs[[1]]
```

Construct sequence table and remove chimeras {#construct-sequence-table-and-remove-chimeras .unnumbered}
--------------------------------------------------------------------------------------------------------
```{r mergers}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose=TRUE)
```

```{r seqtab}
seqtabAll <- makeSequenceTable(mergers)#[!grepl("119blank", names(mergers))])
table(nchar(getSequences(seqtabAll)))
```

```{r chimeras}
seqtabNoC <- removeBimeraDenovo(seqtabAll)
#saveRDS(seqtabNoC, "C:/Users/Ben/Desktop/cornerstone_ITS_fastq/seqtabNoC3.rds")
```
Assign taxonomy {#assign-taxonomy .unnumbered}
----------------------------------------------
```{r}
#seqtabNoC <- readRDS("C:/Users/benja/Desktop/cornerstone_ITS_fastq/seqtabNoC3.rds")
```

```{r}
unite.ref <- "C:/Users/benja/Desktop/cornerstone_files/sh_general_release_dynamic_04_02_2020.fasta"  # CHANGE ME to location on your machine (renamed to avoid issues with the way computers read '.')
taxTab <- assignTaxonomy(seqtabNoC, unite.ref, multithread = FALSE, tryRC = TRUE) #uses reverse strand to improve taxonomy classification if the forward strand is too ambiguous - takes longer but worth it imo. 
#write.table(taxTab, "C:/Users/benja/Desktop/ITS_tax3.tsv", sep = "\t", quote=F, col.names=NA)
```

```{r}
# giving our seq headers more manageable names (ASV_1, ASV_2...)
asv_seqs <- colnames(seqtabNoC)
asv_headers <- vector(dim(seqtabNoC)[2], mode="character")
for (i in 1:dim(seqtabNoC)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep="_")
}
```

```{r}
# making and writing out a fasta of our final ASV seqs:
asv_fasta <- c(rbind(asv_headers, asv_seqs))
#write(asv_fasta, "C:/Users/benja/Desktop/ITS_ASVs3.fa")
```

```{r}
 # count table:
asv_tab <- t(seqtabNoC)
row.names(asv_tab) <- sub(">", "", asv_headers)
#write.table(asv_tab, "C:/Users/benja/Desktop/ITS_ASVs_counts3.tsv", sep="\t", quote=F, col.names=NA)
```

```{r}
# tax table:
  # creating table of taxonomy and setting any that are unclassified as "NA"
ranks <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
#asv_tax <- t(sapply(tax_info, function(x) {
 # m <- match(ranks, x$rank)
 # taxa <- x$taxon[m]
#  taxa[startsWith(taxa, "unclassified_")] <- NA
#  taxa
#}))
colnames(taxTab) <- ranks
#rownames(taxTab) <- gsub(pattern=">", replacement="", x=asv_headers)

#write.table(taxTab, "C:/Users/benja/Desktop/ITS_ASVs_taxonomy4.tsv", sep = "\t", quote=F, col.names=NA)
```

```{r}
#write.table(asv_tax, "C:/Users/benja/Desktop/tax33.tsv", sep = "\t", quote=F, col.names=NA)
#write.table(taxTabTrue, "C:/Users/benja/Desktop/tax44.tsv", sep = "\t", quote=F, col.names=NA)
#write.table(taxTabMult, "C:/Users/benja/Desktop/tax55.tsv", sep = "\t", quote=F, col.names=NA)
```

Construct phylogenetic tree {#construct-phylogenetic-tree .unnumbered}
----------------------------------------------------------------------
```{r msa, output=FALSE,message=FALSE}
seqs <- getSequences(seqtabNoC)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```

```{r tree}
phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) # Note, tip order != (symbol '!=' means 'does not equal' in R) sequence order
fit = pml(treeNJ, data=phangAlign)
fitJC <- optim.pml(fit)
fitJC

```

```{r}
# search for a better tree using NNI rearrangements
  fitJC <- optim.pml(fitJC, optNni=TRUE)
  fitJC
  plot(fitJC$tree)
```

```{r}
# JC + Gamma + I - model
  fitJC_GI <- update(fitJC, k=4, inv=.2)
# optimize shape parameter + proportion of invariant sites
  fitJC_GI <- optim.pml(fitJC_GI, optGamma=TRUE, optInv=TRUE)
  fitJC_GI
```

```{r}
# GTR + Gamma + I - model
  fitGTR <- optim.pml(fitJC_GI, rearrangement = "stochastic",
      optGamma=TRUE, optInv=TRUE, model="GTR")
fitGTR
```

```{r}
detach("package:phangorn", unload=TRUE)
```

Combine data into a phyloseq object {#combine-data-into-a-phyloseq-object .unnumbered}
--------------------------------------------------------------------------------------
```{r samdat}
samdf <- read.csv("C:/Users/benja/Desktop/cornerstone_ITS_fastq/meta2.csv",header=TRUE)
rownames(samdf) <- samdf$SampleID
samdf
```

```{r phyloseqObj}
ps2 <- phyloseq(otu_table(seqtabNoC, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxTab),phy_tree(fitGTR$tree))
ps2 <- prune_samples(sample_names(ps2) != "blank", ps2) # Remove blank sample
ps2
```

```{r}
#saveRDS(ps2, "C:/Users/benja/Desktop/ITS_ps2.rds") #make sure to save this before clearing workspace!!!
```

```{r sessioninfo}
devtools::session_info()
```

