---
title: "cornerstone_16s_dada2_github"
author: Benjamin Robinson
output: html_notebook
---
```{r style, echo=FALSE, message=FALSE, warning=FALSE, results="asis"}
library("BiocStyle")
library("knitr")
library("rmarkdown")
options(width = 98) 
opts_chunk$set(message = FALSE, error = FALSE, warning = FALSE,
               cache = TRUE, fig.width = 8, fig.height = 7)
```
-------------------------------------------------

```{r, message=FALSE, warning=FALSE}
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
   install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
if(any(!.inst)) {
   source("http://bioconductor.org/biocLite.R")
   biocLite(.bioc_packages[!.inst], ask = F)
}
# Load packages into session, and print package version
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
set.seed(100)
```

```{r path}
miseq_path <- "C:/Users/benja/Desktop/cornerstone_16S_fastq/Run4" # CHANGE to the directory containing the fastq files after unzipping.
list.files(miseq_path)
```

Filter and Trim {#trim-and-filter .unnumbered}
----------------------------------------------
```{r filenames}
# Sort ensures forward/reverse reads are in same order
fnFs <- sort(list.files(miseq_path, pattern="_R1_001.fastq"))
fnRs <- sort(list.files(miseq_path, pattern="_R2_001.fastq"))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)
sampleNames <- sapply(strsplit(sampleNames, "202"), `[`, 1)
# Specify the full path to the fnFs and fnRs
fnFs <- file.path(miseq_path, fnFs)
fnRs <- file.path(miseq_path, fnRs)
fnFs[1:3]
fnRs[1:3]
```

The first two forward reads:
```{r see-quality-F}
plotQualityProfile(fnFs[1:2])
```

The first two reverse reads:
```{r see-quality-R}
plotQualityProfile(fnRs[1:2])
```

We define the filenames for the filtered fastq.gz files:
```{r filt-names}
filt_path <- file.path(miseq_path, "filtered") # Place filtered files in filtered/ subdirectory
if(!file_test("-d", filt_path)) dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq.gz"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq.gz"))

```

```{r}
sampleNames
```

**Filter the forward and reverse reads**:
```{r filter, message=FALSE, warning=FALSE}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,160),
              maxN=0, maxEE=c(1,1), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)
```

Infer sequence variants {#infer-sequence-variants .unnumbered}
--------------------------------------------------------------
```{r dereplicate, message=FALSE}
derepFs <- derepFastq(filtFs, verbose=TRUE)
derepRs <- derepFastq(filtRs, verbose=TRUE)
# Name the derep-class objects by the sample names
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames
```

```{r learnErrorRates}
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
```

```{r plotrates, fig.show= "hold", fig.height = 8, fig.cap= "Estimated Error rates (both forward and reverse)"}
plotErrors(errF)
plotErrors(errR)
```

```{r dadaStep}
dadaFs <- dada(derepFs, err=errF, pool="pseudo")
dadaRs <- dada(derepRs, err=errR, pool="pseudo")
```

Inspecting the dada-class object returned by dada:
```{r see-dada}
dadaFs[[1]]
```

Construct sequence table and remove chimeras {#construct-sequence-table-and-remove-chimeras .unnumbered}
--------------------------------------------------------------------------------------------------------
```{r mergers}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, minOverlap=20)
```

```{r seqtab}
seqtabAll <- makeSequenceTable(mergers[!grepl("blank", names(mergers))])
table(nchar(getSequences(seqtabAll)))
```

```{r chimeras}
seqtabNoC <- removeBimeraDenovo(seqtabAll)
#saveRDS(seqtabNoC, "C:/Users/Ben/Desktop/seqtabNoC.rds")
```
Assign taxonomy {#assign-taxonomy .unnumbered}
----------------------------------------------
```{r}
#seqtabNoC <- readRDS("C:/Users/benja/Desktop/CornerstoneseqtabNoC.rds")
sum(seqtabNoC)
```

```{r tax}
fastaRef <- "C:/Users/benja/Desktop/cornerstone_files/silva_nr_v138_train_set.fa.gz"
speciesRef <- "C:/Users/benja/Desktop/cornerstone_files/silva_species_assignment_v138.fa.gz"
taxTab <- assignTaxonomy(seqtabNoC, refFasta = fastaRef, multithread=FALSE, verbose=FALSE)
taxTabTrue <- addSpecies(taxTab, refFasta = speciesRef, verbose = FALSE)
taxTabMult <- addSpecies(taxTab, refFasta = speciesRef, allowMultiple = TRUE, verbose = FALSE)
unname(head(taxTabMult))
```

```{r}
## creating DNAStringSet object of our ASVs
load("C:/Users/benja/Desktop/cornerstone_files/DAIRYdb_v1.2.4_20200603_IDTAXA.rdata") # CHANGE TO THE PATH OF YOUR TRAINING SET
dna <- DNAStringSet(getSequences(seqtabNoC))
IdTaxaDdb <- IdTaxa(test=dna, trainingSet=trainingSet, strand="both", processors=NULL)
```

```{r}
asv_seqs <- colnames(seqtabNoC)
asv_headers <- vector(dim(seqtabNoC)[2], mode="character")
for (i in 1:dim(seqtabNoC)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep="_")
}
```

```{r}
ranks <- c("7", "4", "6", "1", "5", "2", "3", "8")
asv_tax <- t(sapply(IdTaxaDdb, function(x) {
  m <- match(ranks, x$rank)
  taxa <- x$taxon[m]
  taxa[startsWith(taxa, "unclassified_")] <- NA
  taxa
}))
colnames(asv_tax) <- ranks
rownames(asv_tax) <- gsub(pattern=">", replacement="", x=asv_headers)
```

```{r}
rownames(taxTabTrue) <- gsub(pattern=">", replacement="", x=asv_headers)
rownames(taxTabMult) <- gsub(pattern=">", replacement="", x=asv_headers)
```

```{r}
#write.table(asv_tax, "C:/Users/Ben/Desktop/tax1.tsv", sep = "\t", quote=F, col.names=NA)
#write.table(taxTabTrue, "C:/Users/Ben/Desktop/tax2.tsv", sep = "\t", quote=F, col.names=NA)
#write.table(taxTabMult, "C:/Users/Ben/Desktop/tax3.tsv", sep = "\t", quote=F, col.names=NA)
```

Construct phylogenetic tree {#construct-phylogenetic-tree .unnumbered}
----------------------------------------------------------------------
```{r msa, output=FALSE,message=FALSE}
seqs <- getSequences(seqtabNoC)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```

```{r tree}
phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phangAlign)
fitJC <- optim.pml(fit)
fitJC

```

```{r}
# search for a better tree using NNI rearrangements
  fitJC <- optim.pml(fitJC, optNni=TRUE)
  fitJC
  plot(fitJC$tree)
```

```{r}
# JC + Gamma + I - model
  fitJC_GI <- update(fitJC, k=4, inv=.2)
# optimize shape parameter + proportion of invariant sites
  fitJC_GI <- optim.pml(fitJC_GI, optGamma=TRUE, optInv=TRUE)
  fitJC_GI
```

```{r}
# GTR + Gamma + I - model
  fitGTR <- optim.pml(fitJC_GI, rearrangement = "stochastic",
      optGamma=TRUE, optInv=TRUE, model="GTR")
fitGTR
```

```{r}
detach("package:phangorn", unload=TRUE)
```

Combine data into a phyloseq object {#combine-data-into-a-phyloseq-object .unnumbered}
--------------------------------------------------------------------------------------
```{r samdat}
samdf <- read.csv("C:/Users/Ben/Desktop/meta.csv",header=TRUE)
#samdf$SampleID <- paste0(gsub("00", "", samdf$host_subject_id), "D", samdf$age-21)
#samdf <- samdf[!duplicated(samdf$SampleID),] # Remove dupicate entries for reverse reads
#rownames(seqtabAll) <- gsub("124", "125", rownames(seqtabAll)) # Fix discrepancy
#all(rownames(seqtabAll) %in% samdf$SampleID) # TRUE
rownames(samdf) <- samdf$SampleID
#keep.cols <- c("collection_date", "biome", "target_gene", "target_subfragment",
#"host_common_name", "host_subject_id", "age", "sex", "body_product", "tot_mass",
#"diet", "family_relationship", "genotype", "SampleID") 
#samdf <- samdf[rownames(seqtabAll), keep.cols]
samdf
```

```{r phyloseqObj}
ps <- phyloseq(otu_table(seqtabNoC, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxTab),phy_tree(fitGTR$tree))
ps <- prune_samples(sample_names(ps) != "blank", ps) # Remove blank sample
ps
```

```{r}
#saveRDS(ps, "C:/Users/Ben/Desktop/ps.rds")
```

```{r sessioninfo}
devtools::session_info()
```

