---
title: "cornerstone_its_analysis_github"
author: Benjamin Robinson
output: html_notebook
---

```{r style, echo=FALSE, message=FALSE, warning=FALSE, results="asis"}
library("BiocStyle")
library("knitr")
library("rmarkdown")
options(width = 98) 
opts_chunk$set(message = FALSE, error = FALSE, warning = FALSE,
               cache = TRUE, fig.width = 8, fig.height = 7)
```

```{r, message=FALSE, warning=FALSE}
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
   install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
if(any(!.inst)) {
   source("http://bioconductor.org/biocLite.R")
   biocLite(.bioc_packages[!.inst], ask = F)
}
# Load packages into session, and print package version
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
set.seed(100)
```

```{r}
detach("package:phangorn", unload=TRUE)
```

```{r}
## Setup
### Bioconductor and CRAN libraries used
#library(phyloseq)
library(vegan)
library(DESeq2)
library(tidyverse)
library(RColorBrewer)
library(pheatmap)
library(microbiome)
library(ggrepel)
library(dendextend)
library(viridis)
library(reshape)
```

```{r}
ps <- readRDS("C:/Users/benja/Desktop/ITS_ps2.rds")
ps
#summarize_phyloseq(ps.asv)
```

```{r}
meta <- meta(ps)
tax_table <- as.data.frame(tax_table(ps))
otu_table <- as.data.frame(otu_table(ps))
phy_tree <- phy_tree(ps)
```

##Writing tables## - make sure to include # after writing them to prevent accidental overwriting later on!!!!!!!!!!!
```{r}
#write.table(otu_table, "C:/Users/benja/Desktop/ITS_dnas_counts3.tsv",
  #         sep="\t", quote=F, col.names=NA)
#write.table(tax_table, "C:/Users/benja/Desktop/ITS_dnas_taxonomy3.tsv",
 #           sep="\t", quote=F, col.names=NA)
#saveRDS(phy_tree, ("C:/Users/Ben/Desktop/cornerstone_ITS_fastq/ITS_phy_tree2.rds"))

#asv_seqs <- colnames(otu_table) #this contains the ugly raw DNA headers
#asv_headers <- vector(dim(otu_table)[2], mode="character") #this code makes the nicer ones
#for (i in 1:dim(otu_table)[2]) {
#  asv_headers[i] <- paste(">ASV", i, sep="_")
#}

#asv_fasta <- c(rbind(asv_headers, asv_seqs))
#asv_tab <- t(otu_table)
#row.names(asv_tab) <- sub(">", "", asv_headers) #make sure to rename to something different and keep track
#asv_tax <- tax_table
#rownames(asv_tax) <- gsub(pattern=">", replacement="", x=asv_headers)

#write(asv_fasta, "C:/Users/Ben/Desktop/cornerstone_ITS_fastq/ITS_ASVs2.fa")
#write.table(asv_tab, "C:/Users/Ben/Desktop/cornerstone_ITS_fastq/ITS_ASVs_counts2.tsv",
#          sep="\t", quote=F, col.names=NA)
#write.table(asv_tax, "C:/Users/Ben/Desktop/cornerstone_ITS_fastq/ITS_ASVs_taxonomy2.tsv",
#            sep="\t", quote=F, col.names=NA)

```

```{r}
otu_table <- read.table("C:/Users/benja/Desktop/ITS_dnas_counts3.tsv", header=T, row.names=1, check.names=F, sep="\t")
```

```{r}
tax_table <- as.matrix(read.table("C:/Users/benja/Desktop/ITS_dnas_taxonomy3.tsv", header=T,
           row.names=1, check.names=F, sep="\t"))
```

```{r}
#phy_tree = readRDS("C:/Users/Ben/Desktop/ITS_phy_tree.rds")
```

```{r}
count_tab <- read.table("C:/Users/benja/Desktop/ITS_ASVs_counts3.tsv", header=T, row.names=1,
             check.names=F, sep="\t")
```

```{r}
tax_tab <- as.matrix(read.table("C:/Users/benja/Desktop/ITS_ASVs_taxonomy4.tsv", header=T,
           row.names=1, check.names=F, sep="\t"))

```

```{r}
sample_info_tab <- #read.table("C:/Users/Ben/Desktop/ITS_meta.tsv", header=T, row.names=1,
                   #check.names=F, sep="\t")
  read.csv("C:/Users/benja/Desktop/cornerstone_files/ITS_meta.csv", header=TRUE, row.names = "SampleID")

```

  # and setting the color column to be of type "character", which helps later
```{r}
#sample_info_tab$color <- as.character(sample_info_tab$color)
```
  
```{r}
sample_info_tab # to take a peek
```

```{r}
ps.dna <- phyloseq(otu_table(otu_table, taxa_are_rows=FALSE), 
               sample_data(sample_info_tab), 
               tax_table(tax_table), phy_tree(phy_tree))
ps.dna
```

```{r}
ps.asv <- phyloseq(otu_table(count_tab, taxa_are_rows=TRUE), 
               sample_data(sample_info_tab), 
               tax_table(tax_tab))
ps.asv
```

Filtering {#filtering .unnumbered}
---------
```{r taxfilter0}
# Show available ranks in the dataset
rank_names(ps.dna)
rank_names(ps.asv)
# Create table, number of features for each phyla
table(tax_table(ps.dna)[, "Kingdom"], exclude = NULL)
table(tax_table(ps.asv)[, "Kingdom"], exclude = NULL)
```

```{r removeNAphyla}
#ps.dna <- subset_taxa(ps.dna, !is.na(Kingdom) & !Kingdom %in% c("", "uncharacterized"))
#ps.asv <- subset_taxa(ps.asv, !is.na(Kingdom) & !Kingdom %in% c("", "uncharacterized"))
```

```{r}
table(tax_table(ps.dna)[, "Phylum"], exclude = NULL)
table(tax_table(ps.asv)[, "Phylum"], exclude = NULL)
```

```{r}
ps.dna <- subset_taxa(ps.dna, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
ps.asv <- subset_taxa(ps.asv, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
ps.dna
ps.asv
#readRDS(ps.dna, "C:/Users/benja/Desktop/ITS_ps_dna.rds")
#readRDS(ps.asv, "C:/Users/benja/Desktop/ITS_ps_asv.rds")
```

```{r}
summarize_phyloseq(ps.asv)
```

###Pulling subsets for sourcetracker###
```{r}
library(microbiome)
```

```{r}
library(philr); packageVersion("philr")
library(phyloseq); packageVersion("phyloseq")
library(ape); packageVersion("ape")
library(ggplot2); packageVersion("ggplot2")
```

```{r}
#ps.CC<- (subset_samples(ps.asv, location == "PF2"|location=="PF2OrbWeaver"|location == "OrbWeaver"))
#metaCC <- meta(ps.CC)
#tax_table2 <- as.data.frame(tax_table(ps.asv))
#otu_table2 <- as.data.frame(otu_table(ps.asv))
#write.table(otu_table2, "C:/Users/Ben/Desktop/ST_counts_CC.tsv",
#           sep="\t", quote=F, col.names=NA)
#write.table(tax_table2, "C:/Users/Ben/Desktop/ST_taxonomy_CC.tsv",
#            sep="\t", quote=F, col.names=NA)
#write.table(metaCC, "C:/Users/Ben/Desktop/meta_CC.tsv",
#            sep="\t", quote=F, col.names=NA)

```

```{r}
#ps.PH <- (subset_samples(ps.asv, location == "PF3"|location=="PF3OrbWeaver"|location == "OrbWeaver"))
#metaPH <- meta(ps.PH)
#tax_tablePH <- as.data.frame(tax_table(ps.PH))
#otu_tablePH <- as.data.frame(otu_table(ps.PH))
#write.table(otu_tablePH, "C:/Users/Ben/Desktop/ASV_counts_PH.tsv",
#           sep="\t", quote=F, col.names=NA)
#write.table(tax_tablePH, "C:/Users/Ben/Desktop/ASV_taxonomy_PH.tsv",
 #           sep="\t", quote=F, col.names=NA)
#write.table(metaPH, "C:/Users/Ben/Desktop/meta_PH.tsv",
#            sep="\t", quote=F, col.names=NA)

```

```{r}
#ps.BH<- (subset_samples(ps.asv, location == "PF1"|location=="PF1OrbWeaver"|location == "OrbWeaver"))
#metaBH <- meta(ps.BH)
#tax_tableBH <- as.data.frame(tax_table(ps.BH))
#otu_tableBH <- as.data.frame(otu_table(ps.BH))
#write.table(otu_tableBH, "C:/Users/Ben/Desktop/ASV_counts_BH.tsv",
 #          sep="\t", quote=F, col.names=NA)
#write.table(tax_tableBH, "C:/Users/Ben/Desktop/ASV_taxonomy_BH.tsv",
#            sep="\t", quote=F, col.names=NA)
#write.table(metaBH, "C:/Users/Ben/Desktop/meta_BH.tsv",
#            sep="\t", quote=F, col.names=NA)

```

```{r}
Paired <- brewer.pal(12, "Paired") #need the number of colors in palate of choice, plus its color brewer name
Paired_range <- colorRampPalette(Paired) #function that extends it

dark <- brewer.pal(8, "Dark2")
dark_range <- colorRampPalette(dark)
```

```{r}
ps.t<-transform_sample_counts(ps.asv,function(x)x/sum(x))
ps.melt <- (psmelt(ps.t))
ps.melt2 <- ps.melt
ps.melt3 <- ps.melt
ps.melt4 <- ps.melt
#below is the absolutely digusting manner by which I was able to merge taxa into an "other" group strictly for visualization purposes. I specifically did this on the melt object only, since this is only used for ggplot bargraphs in this workflow. no stats were or should be run on data after doing this
#ps.melt$Species[ps.melt$Abundance < 0.01] <- "< 1% abund."
ps.melt$Species[((ps.melt$Species != "Debaryomyces_hansenii"))&
                  ((ps.melt$Species != "Candida_parapsilosis"))&
                  ((ps.melt$Species != "Basidiomycota_Phylum"))&
                  ((ps.melt$Species != "Kluyveromyces_lactis"))&
                  ((ps.melt$Species != "Kluyveromyces_marxianus"))&
                  ((ps.melt$Species != "Trichosporon_lactis"))&
                  ((ps.melt$Species != "Candida_Genus"))&
                  ((ps.melt$Species != "Acremonium_hennebertii"))&
                  ((ps.melt$Species != "Cutaneotrichosporon_curvatus"))&
                  ((ps.melt$Species != "Trichosporon_asahii"))&
                  ((ps.melt$Species != "Cryptococcus_uniguttulatus"))&
                  ((ps.melt$Species != "Yamadazyma_triangularis"))
                ] <- "All Others"
ps.melt
```

```{r}
 taxa.col <- c(
     "All Others"="grey", 
     "Acremonium_hennebertii"="plum2", 
     "Basidiomycota_Phylum"="salmon2", 
     "Candida_Genus"="pink", 
     "Candida_parapsilosis"="slateblue1", 
     "Cryptococcus_uniguttulatus"="black",
     "Cutaneotrichosporon_curvatus"="lightslategrey",
     "Debaryomyces_hansenii"="wheat",
     "Kluyveromyces_lactis"="lavender",
     "Kluyveromyces_marxianus"="powderblue",
     "Trichosporon_asahii"="lightseagreen",
     "Trichosporon_lactis"="seagreen",
     "Yamadazyma_triangularis"="saddlebrown"
     ) 


```

```{r, fig.height=6, fig.width=13.5}
ggplot(ps.melt, aes(x=tag, y = Abundance, fill = Species)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(vjust=0.5, angle = 270),legend.position = "right",axis.title.x =element_blank() ,panel.background = element_blank(), panel.grid=element_blank(), axis.line=element_line('black')) + 
  guides(fill = guide_legend(ncol=1)) +  
  ylab("Relative Abundance (%) \n") + 
  facet_grid(.~facility,switch = "both", scales = "free", space = "free") + 
  scale_fill_manual(values = taxa.col)
```

```{r}
ps.melt2 <- subset(ps.melt2, category == "Rind"|category == "Curd")
#as.character(ps.melt2$day)
ps.melt2$day.ord <- factor(ps.melt2$day, levels = c("0","1","60","120", "180"))
#as.character(ps.melt2$day.ord)
#ps.melt2$Species[ps.melt2$Abundance < 0.01] <- "< 1% abund."
ps.melt2$Species[((ps.melt2$Species != "Debaryomyces_hansenii"))&
                  ((ps.melt2$Species != "Candida_parapsilosis"))&
                  ((ps.melt2$Species != "Basidiomycota_Phylum"))&
                  ((ps.melt2$Species != "Kluyveromyces_lactis"))&
                  ((ps.melt2$Species != "Kluyveromyces_marxianus"))&
                  ((ps.melt2$Species != "Trichosporon_lactis"))&
                  ((ps.melt2$Species != "Candida_Genus"))&
                  ((ps.melt2$Species != "Acremonium_hennebertii"))&
                  ((ps.melt2$Species != "Trichosporon_asahii"))
                ] <- "All Others"
ps.melt2

ps.melt2$Species <- factor(ps.melt2$Species, levels = c("All Others", "Acremonium_hennebertii",     "Basidiomycota_Phylum", "Candida_Genus", "Candida_parapsilosis", "Debaryomyces_hansenii", "Kluyveromyces_lactis", "Kluyveromyces_marxianus", "Trichosporon_asahii", "Trichosporon_lactis"))
     

```

```{r}
 taxa.col2 <- c(
     "All Others"="gray", 
     "Acremonium_hennebertii"="navyblue", 
     "Basidiomycota_Phylum"="skyblue2", 
     "Candida_Genus"="darkseagreen3", 
     "Candida_parapsilosis"="darkseagreen4", 
     "Debaryomyces_hansenii"="lightsalmon",
     "Kluyveromyces_lactis"="lightblue2",
     "Kluyveromyces_marxianus"="lightblue4",
     "Trichosporon_asahii"="steelblue3",
     "Trichosporon_lactis"="steelblue4"
     ) 

```

```{r, fig.height=6, fig.width=13.5}
ggplot(ps.melt2, aes(x=day.ord, y = Abundance, fill = Species)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(vjust=0.5, angle = 270),legend.position = "right",axis.title.x =element_blank() ,panel.background = element_blank(), panel.grid=element_blank(), axis.line=element_line('black')) + 
  guides(fill = guide_legend(ncol=1)) +
  theme(text=element_text(size=15)) +
  ylab("Relative Abundance (%) \n") + 
  facet_grid(.~location,switch = "both", scales = "free", space = "free") + 
  scale_fill_manual(values = taxa.col2)
```

#Environment Bar Graph True
```{r}
ps.melt3 <- subset(ps.melt3, category == "Fan"|category == "Maker"|category == "Form"|category == "Shelf"|category == "Table")

ps.melt3$category.ord <- factor(ps.melt3$category, levels = c("Shelf","Fan","Form", "Maker", "Table"))
#ps.melt3$Species[ps.melt3$Abundance < 0.01] <- "< 1% abund."
ps.melt3$Species[((ps.melt3$Species != "Debaryomyces_hansenii"))&
                  ((ps.melt3$Species != "Candida_parapsilosis"))&
                  ((ps.melt3$Species != "Candida_Genus"))&
                  ((ps.melt3$Species != "Acremonium_hennebertii"))&
                  ((ps.melt3$Species != "Trichosporon_asahii"))
                ] <- "All Others"
ps.melt3

ps.melt3$Species <- factor(ps.melt3$Species, levels = c("All Others", "Acremonium_hennebertii",     "Candida_Genus", "Candida_parapsilosis", "Debaryomyces_hansenii", "Trichosporon_asahii"))

taxa.col3 <- c(
     "All Others"="grey", 
     "Acremonium_hennebertii"="navyblue", 
     "Basidiomycota_Phylum"="skyblue4", 
     "Candida_Genus"="darkseagreen3", 
     "Candida_parapsilosis"="darkseagreen4", 
     "Cryptococcus_uniguttulatus"="sienna",
     "Cutaneotrichosporon_curvatus"="lightslategrey",
     "Debaryomyces_hansenii"="lightsalmon",
     "Kluyveromyces_lactis"="lavender",
     "Kluyveromyces_marxianus"="lavenderblush3",
     "Trichosporon_asahii"="steelblue3",
     "Trichosporon_lactis"="steelblue4"
)


```

```{r, fig.height=6, fig.width=13.5}
yy<- ggplot(ps.melt3, aes(x=category.ord, y = Abundance, fill = Species)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(vjust=0.5, angle = 270),legend.position = "none",axis.title.x =element_blank() ,panel.background = element_blank(), panel.grid=element_blank(), axis.line=element_line('black')) + 
  guides(fill = guide_legend(ncol=1)) +
  theme(text=element_text(size=15)) +
  ylab("Relative Abundance (%) \n") + 
  facet_grid(.~facility,switch = "both", scales = "free", space = "free") + 
  scale_fill_manual(values = taxa.col3)
yy
```

-----------------------------------------------

Rind Beta Diversity Stats True
#make facility tests
```{r}
#CLR
ps.rind <- subset_samples(ps.asv, category == "Rind")
ps_clr_rind <- microbiome::transform(ps.rind, "clr")
#PCA via phyloseq
ord_clr_rind <- phyloseq::ordinate(ps_clr_rind, "RDA")
#Plot scree plot
phyloseq::plot_scree(ord_clr_rind) + 
  geom_bar(stat="identity", fill = "blue") +
  labs(x = "\nAxis", y = "Proportion of Variance\n")
#Examine eigenvalues and % prop. variance explained
head(ord_clr_rind$CA$eig) 
sapply(ord_clr_rind$CA$eig[1:5], function(x) x / sum(ord_clr_rind$CA$eig))     
#Plot ordination
clr1 <- ord_clr_rind$CA$eig[1] / sum(ord_clr_rind$CA$eig)
clr2 <- ord_clr_rind$CA$eig[2] / sum(ord_clr_rind$CA$eig)
ait_make <- phyloseq::plot_ordination(ps_clr_rind, ord_clr_rind, type="samples", color="makefacility", shape = "agingfacility", label = "day") + 
  ggtitle("[PERMANOVA] Pr(<F): 0.001") + 
  labs(col = "Production Facility", shape = "Aging Facility") + 
  geom_point(size = 2) +
  coord_fixed(clr2 / clr1) +
  theme_bw()
ait_make
```

```{r}
#Generate CLR distance matrix
clr_dist_matrix_rind <- phyloseq::distance(ps_clr_rind, method = "euclidean") 
#ADONIS test
vegan::adonis(clr_dist_matrix_rind ~ phyloseq::sample_data(ps_clr_rind)$facility)
anova(betadisper(clr_dist_matrix_rind, phyloseq::sample_data(ps_clr_rind)$facility))
```

```{r, warning=F, message=F}
#PhILR
GP.rind <- subset_samples(ps.dna, category == "Rind")
GP.rind <- transform_sample_counts(GP.rind, function (x) x + 1) #adding a pseudocount of 1 to all values, since log is taken after transformed, this is fine. 
#GP.rind <- microbiome::transform(GP.rind, "compositional")
GP.rind

is.rooted(phy_tree(GP.rind)) # Is the tree Rooted?
tree.rind <- (root(phy_tree(GP.rind), 1, r = TRUE))
is.rooted(tree.rind)
is.binary.tree(tree.rind) # All multichotomies resolved?
GP.rind

tree.rind <- makeNodeLabel(tree.rind, method="number", prefix='n')

name.balance(tree.rind, tax_table(GP.rind), 'n1')

otu.table.rind <- (otu_table(GP.rind))
metadata.rind <- sample_data(GP.rind)
tax.rind <- tax_table(GP.rind)

weight <- philr::calculate.blw(tree.rind)
gp.philr.rind <- philr(otu.table.rind, tree.rind, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist.rind <- dist(gp.philr.rind, method="euclidean")
gp.nmds.rind <- ordinate(GP.rind, 'NMDS', distance=gp.dist.rind)
B <- plot_ordination(GP.rind, gp.nmds.rind, color="makefacility", shape = "agingfacility", label = "day") + 
  geom_point(size=2) + 
  ggtitle("[PERMANOVA] Pr(<F): 0.001") + 
  labs(col = "Production Facility", shape = "Aging Facility") +
  theme_bw()
B
```

```{r}
#ADONIS test
vegan::adonis(gp.dist.rind ~ phyloseq::sample_data(GP.rind)$facility)
#Dispersion test and plot
anova(betadisper(gp.dist.rind, phyloseq::sample_data(GP.rind)$facility))
```

```{r, fig.height=8.5, fig.width=11}
cowplot::plot_grid(ait_make, B, nrow = 2, ncol = 1, scale = 1, labels = c("A", "B"))
```

```{r, warning=F, message=F}
#Phylogeny-plots
ps.rind <- subset_samples(ps.dna, category == "Rind")
ps.rind.rel <- microbiome::transform(ps.rind, "compositional")
#ps.rind <- transform_sample_counts(ps.rind,function(x) log(1 + x))

#Generate distances
ord_bray_rind <- ordinate(ps.rind.rel, method = "NMDS", distance = "bray") 
ord_jsd_rind <- ordinate(ps.rind, method = "NMDS", distance = "jaccard")   
#Plot ordinations
e <- plot_ordination(ps.rind.rel, ord_bray_rind, label = "day", color = "makefacility", shape = "agingfacility") + geom_point(size = 2) + ggtitle("[PERMANOVA] Pr(<F): 0.01") +   labs(col = "Production Facility", shape = "Aging Facility") + theme_bw()  
f <- plot_ordination(ps.rind, ord_jsd_rind, label = "day", color = "makefacility", shape="agingfacility") + geom_point(size = 2) + ggtitle("[PERMANOVA] Pr(<F): 0.18") +   labs(col = "Production Facility", shape = "Aging Facility")+ theme_bw()  
cowplot::plot_grid(e, f, nrow = 1, ncol = 2, scale = .9, labels = c("Bray", "Jaccard"), title = "Rind") 
e
f
```

```{r}
bray_dist_matrix_rind <- phyloseq::distance(ps.rind.rel, method = "bray")
vegan::adonis(bray_dist_matrix_rind ~ phyloseq::sample_data(ps.rind.rel)$facility)
anova(betadisper(bray_dist_matrix_rind, phyloseq::sample_data(ps.rind.rel)$facility))
```

```{r}
jaccard_dist_matrix_rind <- phyloseq::distance(ps.rind, method = "jaccard")
vegan::adonis(jaccard_dist_matrix_rind ~ phyloseq::sample_data(ps.rind)$facility)
anova(betadisper(jaccard_dist_matrix_rind, phyloseq::sample_data(ps.rind)$facility))
```

```{r, warning=F, message=F}
#Phylogeny-plots
ps.rind.phy <- subset_samples(ps.dna, category == "Rind")
ps.rind.phy.rel <- microbiome::transform(ps.rind.phy, "compositional")
#Generate distances
ord_unifrac_rind <- ordinate(ps.rind.phy.rel, method = "NMDS", distance = "wunifrac") 
ord_unifrac_un_rind <- ordinate(ps.rind.phy, method = "NMDS", distance = "unifrac")   
#Plot ordinations
g <- plot_ordination(ps.rind.phy.rel, ord_unifrac_rind, label = "day", color = "makefacility", shape = "agingfacility") + geom_point(size = 2) + ggtitle("[PERMANOVA] Pr(<F): 0.001") +   labs(col = "Production Facility", shape = "Aging Facility") + theme_bw()  
h <- plot_ordination(ps.rind.phy, ord_unifrac_un_rind, label = "day", color = "makefacility", shape="agingfacility") + geom_point(size = 2) + ggtitle("[PERMANOVA] Pr(<F): 0.001") +   labs(col = "Production Facility", shape = "Aging Facility")+ theme_bw()  
cowplot::plot_grid(g, h, nrow = 1, ncol = 2, scale = .9, labels = c("Wunifrac", "Unifrac")) 
g
h
```

```{r, warning=F, message=F}
#weighted unifrac stats
wuni_dist_matrix_rind <- phyloseq::distance(ps.rind.phy.rel, method = "wunifrac") 
vegan::adonis(wuni_dist_matrix_rind ~ phyloseq::sample_data(ps.rind.phy)$makefacility)
anova(betadisper(wuni_dist_matrix_rind, phyloseq::sample_data(ps.rind.phy)$makefacility))
```

```{r, warning=F, message=F}
#unweighted unifrac stats
uni_dist_matrix_rind <- phyloseq::distance(ps.rind.phy, method = "unifrac") 
vegan::adonis(uni_dist_matrix_rind ~ phyloseq::sample_data(ps.rind.phy)$makefacility)
anova(betadisper(uni_dist_matrix_rind, phyloseq::sample_data(ps.rind.phy)$makefacility))
```

```{r, fig.height=7.5, fig.width=15}
cowplot::plot_grid(e, f, g, h, nrow = 2, ncol = 2, scale = .9, labels = c("Bray", "Jaccard", "Wunifrac", "Unifrac"), title = "Rind")
```

#aging facility tests
#BIRCHRUN HILLS
```{r, warning=F, message=F}
#Birchrun Hills
ps.rind.bh <- subset_samples(ps.rind, makefacility == "PF1")
ps.rind.phy.rel.bh <- subset_samples(ps.rind.phy.rel, makefacility == "PF1")
ps_clr_rind.bh <- microbiome::transform(ps.rind.bh, "clr")
#Generate distance matrix
clr_dist_matrix_rind.bh <- phyloseq::distance(ps_clr_rind.bh, method = "euclidean") 
#ADONIS test
vegan::adonis(clr_dist_matrix_rind.bh ~ phyloseq::sample_data(ps_clr_rind.bh)$agingfacility)
#Dispersion test and plot
anova(betadisper(clr_dist_matrix_rind.bh, phyloseq::sample_data(ps_clr_rind.bh)$agingfacility))
```

```{r, warning=F, message=F}
GP.rind.bh <- subset_samples(GP.rind, makefacility == "PF1")

is.rooted(phy_tree(GP.rind.bh)) # Is the tree Rooted?
tree.rind.bh <- (root(phy_tree(GP.rind.bh), 1, r = TRUE))
is.rooted(tree.rind.bh)
is.binary.tree(tree.rind.bh) # All multichotomies resolved?

tree.rind.bh <- makeNodeLabel(tree.rind.bh, method="number", prefix='n')

name.balance(tree.rind.bh, tax_table(GP.rind.bh), 'n1')

otu.table.rind.bh <- (otu_table(GP.rind.bh))
metadata.rind.bh <- sample_data(GP.rind.bh)
tax.rind.bh <- tax_table(GP.rind.bh)

weight <- philr::calculate.blw(tree.rind.bh)
gp.philr.rind.bh <- philr(otu.table.rind.bh, tree.rind.bh, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist.rind.bh <- dist(gp.philr.rind.bh, method="euclidean")
#ADONIS test
vegan::adonis(gp.dist.rind.bh ~ phyloseq::sample_data(GP.rind.bh)$agingfacility)
#Dispersion test and plot
anova(betadisper(gp.dist.rind.bh, phyloseq::sample_data(GP.rind.bh)$agingfacility))
```

```{r, warning=F, message=F}
bray_dist_matrix_rind.bh <- phyloseq::distance(ps.rind.phy.rel.bh, method = "bray")
vegan::adonis(bray_dist_matrix_rind.bh ~ phyloseq::sample_data(ps.rind.phy.rel.bh)$agingfacility)
anova(betadisper(bray_dist_matrix_rind.bh, phyloseq::sample_data(ps.rind.phy.rel.bh)$agingfacility))
```

```{r, warning=F, message=F}
jaccard_dist_matrix_rind.bh <- phyloseq::distance(ps.rind.bh, method = "jaccard")
vegan::adonis(jaccard_dist_matrix_rind.bh ~ phyloseq::sample_data(ps.rind.bh)$agingfacility)
anova(betadisper(jaccard_dist_matrix_rind.bh, phyloseq::sample_data(ps.rind.bh)$agingfacility))
```

```{r, warning=F, message=F}
#weighted unifrac stats
wuni_dist_matrix_rind.bh <- phyloseq::distance(ps.rind.phy.rel.bh, method = "wunifrac") 
vegan::adonis(wuni_dist_matrix_rind.bh ~ phyloseq::sample_data(ps.rind.phy.rel.bh)$agingfacility)
anova(betadisper(wuni_dist_matrix_rind.bh, phyloseq::sample_data(ps.rind.phy.rel.bh)$agingfacility))
```

```{r, warning=F, message=F}
#unweighted unifrac stats
uni_dist_matrix_rind.bh <- phyloseq::distance(ps.rind.bh, method = "unifrac") 
vegan::adonis(uni_dist_matrix_rind.bh ~ phyloseq::sample_data(ps.rind.bh)$agingfacility)
anova(betadisper(uni_dist_matrix_rind.bh, phyloseq::sample_data(ps.rind.bh)$agingfacility))
```

#CATO CORNER
```{r, warning=F, message=F}
#Cato Corner
ps.rind.cc <- subset_samples(ps.rind, makefacility == "PF2")
ps.rind.phy.rel.cc <- subset_samples(ps.rind.phy.rel, makefacility == "PF2")
ps_clr_rind.cc <- microbiome::transform(ps.rind.cc, "clr")
#Generate distance matrix
clr_dist_matrix_rind.cc <- phyloseq::distance(ps_clr_rind.cc, method = "euclidean") 
#ADONIS test
vegan::adonis(clr_dist_matrix_rind.cc ~ phyloseq::sample_data(ps_clr_rind.cc)$agingfacility)
#Dispersion test and plot
anova(betadisper(clr_dist_matrix_rind.cc, phyloseq::sample_data(ps_clr_rind.cc)$agingfacility))
```

```{r, warning=F, message=F}
GP.rind.cc <- subset_samples(GP.rind, makefacility == "PF2")

is.rooted(phy_tree(GP.rind.cc)) # Is the tree Rooted?
tree.rind.cc <- (root(phy_tree(GP.rind.cc), 1, r = TRUE))
is.rooted(tree.rind.cc)
is.binary.tree(tree.rind.cc) # All multichotomies resolved?

tree.rind.cc <- makeNodeLabel(tree.rind.cc, method="number", prefix='n')

name.balance(tree.rind.cc, tax_table(GP.rind.cc), 'n1')

otu.table.rind.cc <- (otu_table(GP.rind.cc))
metadata.rind.cc <- sample_data(GP.rind.cc)
tax.rind.cc <- tax_table(GP.rind.cc)

weight <- philr::calculate.blw(tree.rind.cc)
gp.philr.rind.cc <- philr(otu.table.rind.cc, tree.rind.cc, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist.rind.cc <- dist(gp.philr.rind.cc, method="euclidean")
#ADONIS test
vegan::adonis(gp.dist.rind.cc ~ phyloseq::sample_data(GP.rind.cc)$agingfacility)
#Dispersion test and plot
anova(betadisper(gp.dist.rind.cc, phyloseq::sample_data(GP.rind.cc)$agingfacility))
```

```{r, warning=F, message=F}
bray_dist_matrix_rind.cc <- phyloseq::distance(ps.rind.phy.rel.cc, method = "bray")
vegan::adonis(bray_dist_matrix_rind.cc ~ phyloseq::sample_data(ps.rind.phy.rel.cc)$facility)
anova(betadisper(bray_dist_matrix_rind.cc, phyloseq::sample_data(ps.rind.phy.rel.cc)$facility))
```

```{r, warning=F, message=F}
jaccard_dist_matrix_rind.cc <- phyloseq::distance(ps.rind.cc, method = "jaccard")
vegan::adonis(jaccard_dist_matrix_rind.cc ~ phyloseq::sample_data(ps.rind.cc)$facility)
anova(betadisper(jaccard_dist_matrix_rind.cc, phyloseq::sample_data(ps.rind.cc)$facility))
```

```{r, warning=F, message=F}
#weighted unifrac stats
wuni_dist_matrix_rind.cc <- phyloseq::distance(ps.rind.phy.rel.cc, method = "wunifrac") 
vegan::adonis(wuni_dist_matrix_rind.cc ~ phyloseq::sample_data(ps.rind.phy.rel.cc)$facility)
anova(betadisper(wuni_dist_matrix_rind.cc, phyloseq::sample_data(ps.rind.phy.rel.cc)$facility))
```

```{r, warning=F, message=F}
#unweighted unifrac stats
uni_dist_matrix_rind.cc <- phyloseq::distance(ps.rind.cc, method = "unifrac") 
vegan::adonis(uni_dist_matrix_rind.cc ~ phyloseq::sample_data(ps.rind.cc)$facility)
anova(betadisper(uni_dist_matrix_rind.cc, phyloseq::sample_data(ps.rind.cc)$facility))
```

#PARISH HILL
```{r, message=F, warning=F}
#Parish Hill
ps.rind.ph <- subset_samples(ps.rind, makefacility == "PF3")
ps.rind.phy.rel.ph <- subset_samples(ps.rind.phy.rel, makefacility == "PF3")
ps_clr_rind.ph <- microbiome::transform(ps.rind.ph, "clr")
#Generate distance matrix
clr_dist_matrix_rind.ph <- phyloseq::distance(ps_clr_rind.ph, method = "euclidean") 
#ADONIS test
vegan::adonis(clr_dist_matrix_rind.ph ~ phyloseq::sample_data(ps_clr_rind.ph)$facility)
#Dispersion test and plot
anova(betadisper(clr_dist_matrix_rind.ph, phyloseq::sample_data(ps_clr_rind.ph)$facility))
```

```{r, warning=F, message=F}
GP.rind.ph <- subset_samples(GP.rind, makefacility == "PF3")

is.rooted(phy_tree(GP.rind.ph)) # Is the tree Rooted?
tree.rind.ph <- (root(phy_tree(GP.rind.ph), 1, r = TRUE))
is.rooted(tree.rind.ph)
is.binary.tree(tree.rind.ph) # All multichotomies resolved?

tree.rind.ph <- makeNodeLabel(tree.rind.ph, method="number", prefix='n')

name.balance(tree.rind.ph, tax_table(GP.rind.ph), 'n1')

otu.table.rind.ph <- (otu_table(GP.rind.ph))
metadata.rind.ph <- sample_data(GP.rind.ph)
tax.rind.ph <- tax_table(GP.rind.ph)

weight <- philr::calculate.blw(tree.rind.ph)
gp.philr.rind.ph <- philr(otu.table.rind.ph, tree.rind.ph, 
                  part.weights='enorm.x.gm.counts', 
                  ilr.weights='blw.sqrt')

gp.dist.rind.ph <- dist(gp.philr.rind.ph, method="euclidean")
#ADONIS test
vegan::adonis(gp.dist.rind.ph ~ phyloseq::sample_data(GP.rind.ph)$facility)
#Dispersion test and plot
anova(betadisper(gp.dist.rind.ph, phyloseq::sample_data(GP.rind.ph)$facility))
```

```{r}
bray_dist_matrix_rind.ph <- phyloseq::distance(ps.rind.phy.rel.ph, method = "bray")
vegan::adonis(bray_dist_matrix_rind.ph ~ phyloseq::sample_data(ps.rind.phy.rel.ph)$facility)
anova(betadisper(bray_dist_matrix_rind.ph, phyloseq::sample_data(ps.rind.phy.rel.ph)$facility))
```

```{r}
jaccard_dist_matrix_rind.ph <- phyloseq::distance(ps.rind.ph, method = "jaccard")
vegan::adonis(jaccard_dist_matrix_rind.ph ~ phyloseq::sample_data(ps.rind.ph)$facility)
anova(betadisper(jaccard_dist_matrix_rind.ph, phyloseq::sample_data(ps.rind.ph)$facility))
```

```{r, warning=F, message=F}
#weighted unifrac stats
wuni_dist_matrix_rind.ph <- phyloseq::distance(ps.rind.phy.rel.ph, method = "wunifrac") 
vegan::adonis(wuni_dist_matrix_rind.ph ~ phyloseq::sample_data(ps.rind.phy.rel.ph)$facility)
anova(betadisper(wuni_dist_matrix_rind.ph, phyloseq::sample_data(ps.rind.phy.rel.ph)$facility))
```

```{r, warning=F, message=F}
#unweighted unifrac stats
uni_dist_matrix_rind.ph <- phyloseq::distance(ps.rind.ph, method = "unifrac") 
vegan::adonis(uni_dist_matrix_rind.ph ~ phyloseq::sample_data(ps.rind.ph)$facility)
anova(betadisper(uni_dist_matrix_rind.ph, phyloseq::sample_data(ps.rind.ph)$facility))
```
-----------------------------------------------
```{r sessioninfo}
devtools::session_info()
```

